scripts/kdig

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env bash
+# vim: set syntax=sh:
+
+set -e
+
+if ! hash kubectl 2>/dev/null; then echo "Please install \`kubectl\` before running this command (https://kubernetes.io/docs/tasks/tools/install-kubectl/)"; exit 1; fi
+if ! hash jq 2>/dev/null; then echo "Please install \`jq\` before running this command (https://github.com/stedolan/jq/wiki/Installation)"; exit 1; fi
+
+service=$1
+namespace=$2
+allServices=$(kubectl get services --all-namespaces -o json)
+parseKubectl=".items[] | {externalName: .spec.externalName, name: .metadata.name, namespace: .metadata.namespace, clusterIP: .spec.clusterIP}"
+
+dig () {
+    extractService="$parseKubectl | select(.name=="\""$service"\"") | select(.namespace=="\""$namespace"\"")"
+    extractNamespaces="$parseKubectl | select(.name=="\""$service"\"") | .namespace"
+
+    services=$(echo "$allServices" | jq "$extractService")
+    externalName=$(echo "$services" | jq '. | .externalName')
+    suffix=$(echo $externalName | awk -F. '{print $4}')
+
+    if [ "$suffix" == "cluster" ]; then
+	parentNamespace=$(echo $externalName | awk -F. '{print $2}')
+    else
+	parentNamespace=""
+    fi
+
+    if [ -z "$services" ]; then
+	availableNamespaces=$(echo "$allServices" | jq "$extractNamespaces")
+	if [ -z "$availableNamespaces" ]; then
+	    echo "Service $service does not exist in this kube context"
+	    exit 0
+	fi
+
+	echo -e "Service $service is not resolvable from $namespace. It exists as a service in namespaces:\n$availableNamespaces"
```

Comment:
"not resolvable from namespace $namespace"  and "exists as a service in the following namespaces:\n..."

Reply:
actually this is fine